src/core/time.ts

```diff
@@ -126,7 +139,14 @@ export class TimeValue {
       seconds += 1;
       nanoseconds -= TimeUnit.sec;
     }
-    return TimeValue.secsAndNs(seconds, nanoseconds);
+
+    // Check an overflow.
+    if (!Number.isSafeInteger(seconds)) {
+      // The overflow happens.
+      return TimeValue.forever();
```

Comment:
We currently don't allow negative time intervals. If we decide to allow them at some point, we have to pay attention to the direction of the overflow, but for now, this works.